modules/neovim/init/highlight.nix

```
@@ -0,0 +1,121 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  inherit (lib.options) mkOption literalExpression;
+  inherit (lib.types) nullOr attrsOf listOf submodule bool ints str enum;
+  inherit (lib.strings) hasPrefix concatStringsSep;
+  inherit (lib.attrsets) mapAttrsToList;
+  inherit (lib.nvim.dag) entryAnywhere;
+  inherit (lib.nvim.lua) toLuaObject;
+  inherit (lib.nvim.types) hexColor;
+
+  mkColorOption = target:
+    mkOption {
+      type = nullOr hexColor;
+      default = null;
+      description = ''
+        The ${target} color to use. Written as color name or hex "#RRGGBB".
+      '';
+      example = "#ebdbb2";
+    };
+
+  mkBoolOption = name:
+    mkOption {
+      type = nullOr bool;
+      default = null;
+      description = ''Whether to enable ${name}'';
```

```suggestion
      description = "Whether to enable ${name}";
```

Also move `example` between `default` and `description` please.